app/javascript/mastodon/components/button/index.tsx

```diff
@@ -42,28 +46,46 @@ export const Button: React.FC<Props> = ({
 }) => {
   const handleClick = useCallback<React.MouseEventHandler<HTMLButtonElement>>(
     (e) => {
-      if (!disabled && onClick) {
+      if (disabled || loading) {
+        e.stopPropagation();
+        e.preventDefault();
+      } else if (onClick) {
         onClick(e);
       }
```

Comment:
why isn't this preventing default and stopping propagation always?

Reply:
It probably wouldn't do much harm but I'd expect a `Button` to be as thin a wrapper around the native `<button>` element as possible, so doing anything extra here wouldn't feel right.

Even the `disabled` check shouldn't strictly have been necessary as `onClick` should never be called for a properly disabled button. I believe the opt-out is only really needed for the new `loading` prop, and previously the `onClick` prop could've just been passed through untouched.